### ISSUE TITLE
Stoploss or Trailing StopLoss in Market

### ISSUE THREAD

**Issue by mkajnar** [freqtrade/freqtrade#10414]
Operating system:
    Debian GNU/Linux 12 (bookworm)

Python Version:
    Python 3.12.4

CCXT version:
   4.3.46

Freqtrade Version: 
   freqtrade docker-2024.6-dev-44a37d11

Describe the problem:

Stoploss is often not applied or is applied with a significant delay on Bybit Exchange even though I have set it to execute at Market. When Trailing Stop Loss is used, it somehow "overlaps the functionality" of Stop Loss, and the Trailing Stop Loss often results in a red position even though it was triggered in a green position. Simply put, Stoploss and Trailing Stoploss are very delayed or completely ignored.

Steps to reproduce:

Create an order in short or long.

If the market turns against me, I want to exit at Stoploss or Trailing Stop Loss.

The position remains, for example, hanging at -20% even though StopLoss is set at 5%.

Observed Results:

The bot incurs a completely unnecessary loss, but it might be sufficient to adjust some parameters to execute the stoploss in a timely manner.

What did you expect to happen?

The bot should immediately apply StopLoss or Trailing StopLoss when set to Market.

Relevant code exceptions or logs:

No relevant code exceptions or logs; the issue occurs without any errors, I simply observe the bot in a losing position.

**Comment by mkajnar** [comment]
![image](https://github.com/freqtrade/freqtrade/assets/5566514/f5e270d5-d041-4b9c-86e0-7b69adba9a61)

Example, stoploss is in the chart visible, but Freqtrade (or Exchange) ignore it (twice)

**Comment by xmatthias** [comment]
Unfortunately, from looking at a chart screenshot and a tale (the description) - we can't derive anything actionable.

I'm trading with stops on bybit all the time (both long and short) - and my stops do trigger and fill just fine.

for us to even try and look at this, please provide:

* Logs from the time of the trade (ideally the full logs from the entry until at least the stop trigger point, ideally until it's actually closed)
* The content of the "order_types" key in the config
* the content of the `custom_stoploss()` function (if you're using this and have `use_custom_stoploss = True`)
* The trade object (see below how to get that)


`freqtrade show-trades --db-url sqlite:///<trades.sqlite> --print-json --trade-ids <tradeid>` 
Please use a "details block" to keep the issue itself kindof readable (see below for details on how to do that). 

<details>
<summary>Expand to see how to do details blocks</summary>

```
<details>
<summary>Expand full </summary>
(wrap in tripple ticks `)
Trades output ...
</details>
```
</details>

**Comment by mkajnar** [comment]
I found this: I had use_order_book enabled, and that was probably the cause. Without it, stop loss is fired just in the time. :)

**Comment by xmatthias** [comment]
`use_order_book` has nothing to do with how orders fill.
that's essentially "selecting" how the price is determined. 
If your order then fired and filled "in time" - then you've simply been lucky this time.

without `use_order_book` - freqtrade is (forced to) use pricing from the "fetch_ticker" endpoint - which does not necessarily show uptodate information - but can show pricing information that's delayed by a few seconds.

**Comment by mkajnar** [comment]
> `use_order_book` has nothing to do with how orders fill. that's essentially "selecting" how the price is determined. If your order then fired and filled "in time" - then you've simply been lucky this time.
> 
> without `use_order_book` - freqtrade is (forced to) use pricing from the "fetch_ticker" endpoint - which does not necessarily show uptodate information - but can show pricing information that's delayed by a few seconds.

You are right, I set it back to use_order_book and set top for 5 (old value was 1)
